api/pacman-api-asset/src/main/java/com/tmobile/pacman/api/asset/Utils/AssetUtils.java

```diff
@@ -0,0 +1,139 @@
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static com.tmobile.pacman.api.commons.Constants.EC2;
+
+@Component
+public class AssetUtils {
+
+    static String CLOUD_QUALYS_POLICY = "Ec2InstanceScannedByQualys_version-1";
+    static String CLOUD_QUALYS_POLICY_RULES = "Ec2WithSeverityVulnerability_version-1";
+
+    private static List<String> statesForEc2 = Arrays.asList(Constants.RUNNING, AssetConstants.STOPPED, AssetConstants.STOPPING);
+
+    public static String fetchInputValFromFilter(Object obj) {
+        String filterExempted;
+        if (obj instanceof String) {
+           return (String) obj;
+        } else if (obj instanceof List) {
+            List<String> filterExemptedList = (List<String>) obj;
+            return (!CollectionUtils.isEmpty(filterExemptedList) && filterExemptedList.size() == 1) ? filterExemptedList.get(0) : "";
+        } else {
+            return  "";
+        }
+    }
+
+    /**
+     * Fetches the intersection of the two input resource ids provided
+     * if there is already an existing set of resource ids to be filtered we
+     * need tp find the intersection of the two, if not we can return the new set of resource ids
+     * if no common resource id exist, then return empty list signifying no data present for both the filter applied     *
+     *
+     * @param existingMust the existing set of resource ids that need to be filteres
+     * @param newAdd       the next set of resource ids that needs to be applied as part of filter     *
+     * @return list common resource ids.
+     */
+    public static List<String> getMustResourceIdsForAssets(List<String> existingMust, List<String> newAdd) {
+        if (CollectionUtils.isEmpty(newAdd)) {
+            return new ArrayList<>();
+        } else if (CollectionUtils.isEmpty(existingMust)) {
+            existingMust.addAll(newAdd);
+        } else {
+            existingMust.retainAll(newAdd);
+            if (existingMust.isEmpty()) {
+                return new ArrayList<>();
+            }
+        }
+        return existingMust;
+    }
+
+    /**
+     * Creates an OR relation in between two parts of es query, one for the asset type ec2 if present in the filter or input
+     * and the other part for rest of the asset type
+     * this is done to apply the state filter for ec2 asset types only     *
+     *
+     * @param targetTypes   target type
+     * @param policyIds     input filter policyId
+     * @param resourceIds   target type
+     * @param qualysEnabled boolean
+     * @return OR object for asset type ec2 and rest.
+     */
+    public static Map<String, Object> createShouldObjectForAssetTypeESQuery(List<String> targetTypes, List<String> policyIds, List<String> resourceIds, boolean qualysEnabled) {
+        List<String> entityTypes = targetTypes.stream().filter(x -> !x.equalsIgnoreCase(EC2)).collect(Collectors.toList());
+        boolean isEc2 = targetTypes.stream().anyMatch(x -> x.equalsIgnoreCase(EC2));
```

Comment:
you just got all NOT EC2 instances above. Can you use this info and do not use stream here?